src/lib.rs

```diff
@@ -162,6 +163,53 @@ pub fn metadata<P: AsRef<Path> + Into<PathBuf>>(path: P) -> io::Result<fs::Metad
     fs::metadata(path.as_ref()).map_err(|source| Error::new(source, ErrorKind::Metadata, path))
 }
 
+/// Wrapper for [`fs::canonicalize`](https://doc.rust-lang.org/stable/std/fs/fn.canonicalize.html).
+pub fn canonicalize<P: AsRef<Path> + Into<PathBuf>>(path: P) -> io::Result<PathBuf> {
+    fs::canonicalize(path.as_ref())
+        .map_err(|source| Error::new(source, ErrorKind::Canonicalize, path))
+}
+
+/// Wrapper for [`fs::hard_link`](https://doc.rust-lang.org/stable/std/fs/fn.hard_link.html).
+pub fn hard_link<P: AsRef<Path> + Into<PathBuf>, Q: AsRef<Path> + Into<PathBuf>>(
+    src: P,
+    dst: Q,
+) -> io::Result<()> {
+    fs::hard_link(src.as_ref(), dst.as_ref())
+        .map_err(|source| SourceDestError::new(source, SourceDestErrorKind::HardLink, src, dst))
+}
+
+/// Wrapper for [`fs::read_link`](https://doc.rust-lang.org/stable/std/fs/fn.read_link.html).
+pub fn read_link<P: AsRef<Path> + Into<PathBuf>>(path: P) -> io::Result<PathBuf> {
+    fs::read_link(path.as_ref()).map_err(|source| Error::new(source, ErrorKind::ReadLink, path))
+}
+
+/// Wrapper for [`fs::rename`](https://doc.rust-lang.org/stable/std/fs/fn.rename.html).
+pub fn rename<P: AsRef<Path> + Into<PathBuf>, Q: AsRef<Path> + Into<PathBuf>>(
+    from: P,
+    to: Q,
+) -> io::Result<()> {
+    fs::rename(from.as_ref(), to.as_ref())
+        .map_err(|source| SourceDestError::new(source, SourceDestErrorKind::Rename, from, to))
+}
+
+/// Wrapper for [`fs::soft_link`](https://doc.rust-lang.org/stable/std/fs/fn.soft_link.html).
+#[deprecated = "replaced with std::os::unix::fs::symlink and \
```

Comment:
Wrapping a deprecated API feels a bit wrong to me. Perhaps we should wrap the platform-specific APIs instead?

Reply:
That's just so the crate is truly a drop-in replacement. I was planning to wrap the os-specific calls as well as I need them myself.